Services/UICore/classes/class.ilGlobalTemplate.php

```diff
@@ -0,0 +1,1998 @@
+	 * @param bool $a_top_only
+	 */
+	public function addAdminPanelToolbar(ilToolbarGUI $toolb,$a_bottom_panel = true, $a_arrow = false)
+	{
+		$this->admin_panel_commands_toolbar = $toolb;
+		$this->admin_panel_arrow = $a_arrow;
+		$this->admin_panel_bottom = $a_bottom_panel;
+	}
+	
+	/**
+	* Put admin panel into template:
+	* - creation selector
+	* - admin view on/off button
+	*/
+	private function fillAdminPanel()
+	{
+		global $DIC;
+		$lng = $DIC->language();
+
+		if ($this->admin_panel_commands_toolbar === null) {
+			return;
+		}
+
+		$toolb = $this->admin_panel_commands_toolbar;
+		assert($toolbar instanceof \ilToolbarGUI);
+
+		// Add arrow if desired.
+		if($this->admin_panel_arrow)
+		{
+			$toolb->setLeadingImage(ilUtil::getImagePath("arrow_upright.svg"), $lng->txt("actions"));
+		}
+
+		$this->fillPageFormAction();
+
+		// Add top admin bar.
+		$this->setCurrentBlock("adm_view_components");
+		$this->setVariable("ADM_PANEL1", $toolb->getHTML());
+		$this->parseCurrentBlock();
+		
+		// Add bottom admin bar if user wants one.
+		if ($this->admin_panel_bottom)
+		{
+			$this->setCurrentBlock("adm_view_components2");
+
+			// Replace previously set arrow image.
+			if ($this->admin_panel_arrow)
+			{
+				$toolb->setLeadingImage(ilUtil::getImagePath("arrow_downright.svg"), $lng->txt("actions"));
+			}
+
+			$this->setVariable("ADM_PANEL2", $toolb->getHTML());
+			$this->parseCurrentBlock();
+		}
+	}
+	
+	public function setPermanentLink($a_type, $a_id, $a_append = "", $a_target = "", $a_title = "")
+	{
+		$this->permanent_link = array(
+			"type" => $a_type,
+			"id" => $a_id,
+			"append" => $a_append,
+			"target" => $a_target,
+			"title" => $a_title);
+
+	}
+	
+	/**
+	* Fill in permanent link
+	*/
+	private function fillPermanentLink()
+	{
+		if (is_array($this->permanent_link))
+		{
+			include_once("./Services/PermanentLink/classes/class.ilPermanentLinkGUI.php");
+			$plinkgui = new ilPermanentLinkGUI(
+				$this->permanent_link["type"],
+				$this->permanent_link["id"],
+				$this->permanent_link["append"],
+				$this->permanent_link["target"]);
+			if ($this->permanent_link["title"] != "")
+			{
+				$plinkgui->setTitle($this->permanent_link["title"]);
+			}
+			$this->setVariable("PRMLINK", $plinkgui->getHTML());
+		}
+	}
+
+	/**
+	 * Reset all header properties: title, icon, description, alerts, action menu
+	 */
+	public function resetHeaderBlock($a_reset_header_action = true)
+	{
+		$this->setTitle(null);
+		$this->setTitleIcon(null);
+		$this->setDescription(null);
+		$this->setAlertProperties(array());		
+		$this->enableDragDropFileUpload(null);
+		
+		// see setFullscreenHeader()
+		if($a_reset_header_action)
+		{
+			$this->setHeaderActionMenu(null);
+		}
+	}	
+	
+	/**
+	 * Enables the file upload into this object by dropping a file.
+	 */
+	public function enableDragDropFileUpload($a_ref_id)
+	{
+		$this->enable_fileupload = $a_ref_id;
+	}
+
+
+	// TEMPLATING AND GLOBAL RENDERING
+	//
+	// Forwards to ilTemplate-member.
```

Comment:
I think this could be a general pattern to clean up that class even further: Factor out functionality into new services (e.g. `RessourceRegistry` for CSS- and JS-files). Keep `ilGlobalTemplate` as a facade. Encourage people to use the new services directly. Then remove `ilGlobalTemplate` at some point.

I would expect this specific  forwards to ` ilTemplate` to become more and more useless when the UI-framework and the new GlobalScreen-service and the UI-framework gain more traction.